src/omv/imlib/draw.c

```diff
@@ -722,6 +722,9 @@ void imlib_draw_row_setup(imlib_draw_row_data_t *data) {
 
     int alpha = data->alpha, max = 256;
 
+    // To avoid having to divide by 255 scale alpha to 0-256 so we can right shift by 8.
```

Comment:
This doesn't look right. You're still dividing by 255, so where's the right shift by 8 ? Also `smuad_alpha` will be `0x1000000` if alpha == 255.

Reply:
It's used like this:

`__SMUAD(smuad_alpha, smuad_pixel) >> 8`

Where `smuad_alpha` is `((alpha << 16) | (256 - alpha))`

It performs `(alpha * src_pixel) + ((256 - alpha) * dst_pixel) / 256`.

Alpha of 255 needs to produce 100% of the source pixel. And alpha of 0 needs to be 100% of the destination pixel.

Reply:
It's possible to divide by 255 instead of 256, but, this can't be done in a single shift. But, it's more work per pixel. So, by doing `alpha = fast_roundf((alpha * 256) / 255.0f);` you save a lot of time per pixel.